File: backend/app/schemas/watchlist.py
from datetime import datetime

from pydantic import BaseModel, Field


class WatchlistItemCreate(BaseModel):
    symbol: str = Field(min_length=1, max_length=16)


class WatchlistItemOut(BaseModel):
    symbol: str
    added_at: datetime
    name: str | None = None
    last_price: float | None = None
    prev_close: float | None = None
    change_pct: float | None = None

    model_config = {"from_attributes": True}

File: frontend/src/api/watchlist.ts
import { api } from "./client";

export type WatchlistItem = {
  symbol: string;
  added_at: string;
  name: string | null;
  last_price: number | null;
  prev_close: number | null;
  change_pct: number | null;
};

export const watchlistApi = {
  list: () => api.get<WatchlistItem[]>("/api/watchlist"),
  add: (symbol: string) => api.post<WatchlistItem>("/api/watchlist", { symbol }),
  remove: (symbol: string) => api.del<void>(`/api/watchlist/${symbol}`),
};
